crates/sui-framework/packages/sui-framework/sources/event.move

```diff
@@ -35,4 +35,14 @@ module sui::event {
     /// The type `T` is the main way to index the event, and can contain
     /// phantom parameters, eg `emit(MyEvent<phantom T>)`.
     public native fun emit<T: copy + drop>(event: T);
+
+
+    #[test_only]
+    /// Get the total number of events emitted during execution so far
+    public native fun num_events(): u32;
+
+    #[test_only]
+    /// Get all events of type `T` emitted during execution.
+    /// Can only be used in testing,
+    public native fun get_events_by_type<T: copy + drop>(): vector<T>;
```

Comment:
Probably just `events_by_type`. But I'm wondering if we should add this to test scenario for discoverability. Not sure

Reply:
I think it probably makes sense here--the `event` module is where I would be inclined to look for a function related to testing events, and this is definitely useful outside of code that uses `test_scenario` (e.g., unit tests that call complex third-party code that emits events + want to check that they look as expected).

On naming: agreed, shorter is always better!